python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -1185,6 +1185,20 @@ def _replace_closurevars(
     )
 
 
+def is_proxy_object(obj: Any) -> bool:
+    """Determine if an object is proxy object
+
+    Parameters
+    ----------
+    obj : object
+        Any python object.
+
+    """
+    if type(type(obj)) == _FastSlowProxyMeta:
+        return True
```

Comment:
This isn't quite general enough because there are some proxy objects that need to check composition of metaclasses. Consider e.g. [the `AbstractHolidayCalendar`](https://github.com/rapidsai/cudf/blob/branch-24.08/python/cudf/cudf/pandas/_wrappers/pandas.py#L1042). You'll have to check for membership in the MRO:
```
>>> import cudf.pandas
>>> cudf.pandas.install()
>>> import pandas as pd
>>> type(pd.ExcelWriter)
<class 'types.ExcelWriter_Meta'>
>>> type(pd.ExcelWriter).__mro__
(<class 'types.ExcelWriter_Meta'>, <class 'abc.ABCMeta'>, <class 'cudf.pandas.fast_slow_proxy._FastSlowProxyMeta'>, <class 'type'>, <class 'object'>)
>>> type(pd.tseries.holiday.AbstractHolidayCalendar)
<class 'types.AbstractHolidayCalendar_Meta'>
>>> type(pd.tseries.holiday.AbstractHolidayCalendar).__mro__
(<class 'types.AbstractHolidayCalendar_Meta'>, <class 'pandas.tseries.holiday.HolidayCalendarMetaClass'>, <class 'cudf.pandas.fast_slow_proxy._FastSlowProxyMeta'>, <class 'type'>, <class 'object'>)
>>> type(pd.tseries.holiday.AbstractHolidayCalendar) == cudf.pandas.fast_slow_proxy._FastSlowProxyMeta
False
>>> cudf.pandas.fast_slow_proxy._FastSlowProxyMeta in type(pd.tseries.holiday.AbstractHolidayCalendar).__mro__
True
```

Reply:
Ah right, some types inherit from multiple metaclasses. Thanks!